resources/design-document.md

```diff
@@ -1,75 +1,197 @@
+* Automatic warehouse layout creation and display
 
-_Define the data models your service will expose in its responses via your *`-Model`* package. These will be equivalent to the *`PlaylistModel`* and *`SongModel`* from the Unit 3 project._
+# 5. Proposed Architecture Overview
 
-## 6.2. _First Endpoint_
+The initial product will provide the minimum lovable product (MLP) including creating, updating, and deleting inventory items, updating their metrics, spinning up new warehouses, selecting their products and arranging the items on the available shelf-space.
 
-_Describe the behavior of the first endpoint you will build into your service API. This should include what data it requires, what data it returns, and how it will handle any known failure cases. You should also include a sequence diagram showing how a user interaction goes from user to website to service to database, and back. This first endpoint can serve as a template for subsequent endpoints. (If there is a significant difference on a subsequent endpoint, review that with your team before building it!)_
+This will be accomplished using a RESTful API and AWS Lambda to create endpoints.
 
-_(You should have a separate section for each of the endpoints you are expecting to build...)_
+Inventory item and warehouse data will be stored in DynamoDB
+
+PopStock will also provide a web interface for users to manage their warehouses and inventory.  There will be a main page showing existing warehouses, with buttons to create a new one or modify existing inventory items. The inventory item page will have buttons linking to pages to create, update, or delete existing inventory items. Clicking the button for an existing warehouse will lead to a page where you can update a warehouse's inventory, run reports, or generate a new layout.  
+# 6. API
+
+## 6.1. Public Models
+```
+//WarehouseModel 
+String itemId;
+String name;
+int region;
+Map<ItemModel, Integer> inventory
+int size
+```
+
+```
+//ItemModel
+String warehouseId;
+String name;
+int regionalDemand;
+double salesForecast
+double perPallet;
+double weight;
+double purchaseCost;
+double baseMargin;
+double rateOfReplenishment;
+String category
+String synergy
+boolean active
+```
+
+```
+//shippedModel
+String itemId // partition key, string
+String warehouseId, sort key, string
+Date date
+int count
+String customerId
+```
+
+## 6.2. Create Item Endpoint
+
+* Accepts POST requests to /items
+* Accepts input data and user info to create a new item and returns an itemModel for the logged-in user
+* 
+## 6.3 Get Item Endpoint
+* Accepts GET requests to /items/id
+* Accepts an itemId and user info and returns the relevant itemModel
+
+## 6.4 Remove Item Endpoint
+* Accepts PUT requests to /items/id/remove
+* Accepts an itemID, sets active to false for the given item, and returns the updated itemModel for the 
```

Comment:
"for the given _item_"

The idea here is that the item may be removed from consideration for stocking calculations (simulating discontinuation), but that there may still be stock available in certain warehouses so it would need to be set to inactive rather than deleted so that the item data may still be accessed..

Really though, this shouldn't be an endpoint, because the PUT endpoint can just be used to change the active boolean, so I'm removing this one altogether.

Reply:
Hahaha, I guess I was more tired than I realized. Great insight, though, about using PUT.